app/workers/proxy_config_affecting_change_worker.rb

```diff
@@ -1,10 +1,23 @@
 # frozen_string_literal: true
 
+discard_on ActiveRecord::RecordNotFound
+
 class ProxyConfigAffectingChangeWorker < ApplicationJob
   def perform(event_id)
     event = EventStore::Repository.find_event!(event_id)
     proxy = Proxy.find(event.proxy_id)
     proxy.affecting_change_history.touch
-  rescue ActiveRecord::RecordNotFound
+  rescue ActiveRecord::StatementInvalid
+    # logging to help us understand the problem
+    proxy.reload
+    proxy_config_affecting_change = proxy..send(:proxy_config_affecting_change)
+    parameters = {
+      proxy_created_at: proxy.created_at,
+      proxy_updated_at: proxy.updated_at,
+      proxy_affecting_change_created_at: proxy_config_affecting_change&.created_at,
+      proxy_affecting_change_updated_at: proxy_config_affecting_change&.updated_at
+    }
+    System::ErrorReporting.report_error(exception, parameters: parameters)
+    retry
```

Comment:
How about `raise` and let the job to be retried as usual?

Reply:
right, good point 💪